File: Televentas Java/Clases/TarjetaCredito.java
package Clases;

import java.time.LocalDate;

public class TarjetaCredito extends Pago {

    private String numero;
    private String titular;
    private LocalDate fechaVencimiento;
    private String cvv;

    public TarjetaCredito(int idPago, double monto, LocalDate fecha,
                          String numero, String titular,
                          LocalDate fechaVencimiento, String cvv) {
        super(idPago, monto, fecha);
        this.numero = numero;
        this.titular = titular;
        this.fechaVencimiento = fechaVencimiento;
        this.cvv = cvv;
    }

    @Override
    public boolean procesar() {
        System.out.println("Pago procesado correctamente");
        return true;
    }

    @Override
    public String obtenerEstado() {
        return "Aprobado";
    }

    public boolean validar() {
        System.out.println("Tarjeta validada correctamente");
        return true;
    }

    public boolean autorizar() {
        System.out.println("Pago autorizado");
        return true;
    }
}

File: TeleVentas/Clases/tarjeta_credito.py
"""Módulo que define la clase TarjetaDeCredito."""

from TeleVentas.Clases.pago import Pago
from datetime import date


class TarjetaDeCredito(Pago):
    """Representa un pago con tarjeta de crédito."""

    def __init__(self, id_pago: int, monto: float,
                 fecha: date, numero: str, titular: str,
                 fecha_vencimiento: date, cvv: str):
        super().__init__(id_pago, monto, fecha)
        self.__numero = numero
        self.__titular = titular
        self.__fecha_vencimiento = fecha_vencimiento
        self.__cvv = cvv

    def procesar(self) -> bool:
        print("Pago procesado correctamente")
        return True

    def obtener_estado(self) -> str:
        return "Aprobado"

    def validar(self) -> bool:
        print("Tarjeta validada correctamente")
        return True

    def autorizar(self) -> bool:
        print("Pago autorizado")
        return True
